### ISSUE TITLE
Failing to use description for summary when neither detail nor summary are provided

### ISSUE THREAD

**Issue by radanskoric** [ruby-grape/grape-swagger#734]
Right now, if an endpoint doesn't declare anything beyond the default `desc` value, the summary field will be empty.

I.e. if this is done:
```
desc 'A short endpoint description'
get '/endpoint' do
end
```

The swagger documentation for this endpoint will not contain the summary field. This is a problem because if you then render using something like [Swagger UI](https://github.com/swagger-api/swagger-ui) the resulting rendering will not contain any short description, the line will have just the url, like this:

<img width="657" alt="screen shot 2018-12-13 at 09 50 34" src="https://user-images.githubusercontent.com/1065097/49926758-99a78700-febc-11e8-885e-b9f0d7f0059a.png">

This seems to me like undesirable behaviour since the endpoint would look more informative if it used the `desc` value as summary.

This behaviour is right now encoded in specs, although indirectly indirectly. There are many specs but one of the simpler cases is in the [mounted_target_class_spec](https://github.com/ruby-grape/grape-swagger/blob/02ede028a3cfe6facc17207af4071f76ba3df698/spec/swagger_v2/mounted_target_class_spec.rb) which declares an endpoint [with just desc](https://github.com/ruby-grape/grape-swagger/blob/02ede028a3cfe6facc17207af4071f76ba3df698/spec/swagger_v2/mounted_target_class_spec.rb#L10) but then doesn't expect any summary [in the result](https://github.com/ruby-grape/grape-swagger/blob/02ede028a3cfe6facc17207af4071f76ba3df698/spec/swagger_v2/mounted_target_class_spec.rb#L42-L46).

If I had to guess I'd guess it didn't used to be like this since the code [tries to read the desc](https://github.com/ruby-grape/grape-swagger/blob/02ede028a3cfe6facc17207af4071f76ba3df698/lib/grape-swagger/endpoint.rb#L142) attribute of route options and use it as summary but current version of grape never sets that attribute anywhere in the code.

Is this really desirable behaviour or a bug?

**Comment by dblock** [comment]
I am surprised `desc` is not used! So I think you should try a PR that fixes this.

**Comment by radanskoric** [comment]
Great, thanks! 

I'll prepare a PR when I find a bit of time. The general idea is to basically just use description if there is no summary option on the route. Does that make sense to you?

The fix is simple, most of the change will be updates to all the specs indirectly encoding this behaviour
in the expected output.

**Comment by arseny-emchik** [comment]
It looks like it depend on `detail` option in `desc` block.
For example:
```ruby
          desc 'List subscriptions' do
            params some
            success some
          end
```
<img width="787" alt="Снимок экрана 2019-07-24 в 13 46 51" src="https://user-images.githubusercontent.com/5054656/61788054-cd1a9e80-ae19-11e9-8d4c-7979b458285e.png">


```ruby
          desc 'List subscriptions' do
            detail ""
            params some
            success some
          end
```
<img width="769" alt="Снимок экрана 2019-07-24 в 13 47 14" src="https://user-images.githubusercontent.com/5054656/61788102-e91e4000-ae19-11e9-844a-10a1a0586cbb.png">


So it works if `detail` option was provided. Even if `detail` option was blank.

Maybe this will help.

**Comment by arseny-emchik** [comment]
@radanskoric 
I've just looked through the code of the gem.
So It looks like a bad UI/UX, but not a bug.

1. There is no option `desc`. So this line of code will never be executed:
https://github.com/ruby-grape/grape-swagger/blob/02ede028a3cfe6facc17207af4071f76ba3df698/lib/grape-swagger/endpoint.rb#L142
1. https://github.com/ruby-grape/grape-swagger/blob/02ede028a3cfe6facc17207af4071f76ba3df698/lib/grape-swagger/endpoint.rb#L143 - This mean that they show description only if there are additional details. So if you provide details, your description will be shown like a summary. Otherwise you need to open tab and see into it. Your description will be marked as `implementation notes` or something like that. You can see it in the line https://github.com/ruby-grape/grape-swagger/blob/02ede028a3cfe6facc17207af4071f76ba3df698/lib/grape-swagger/endpoint.rb#L150

**Comment by radanskoric** [comment]
You're right, that it is the issue. I called it a Bug because I view it as undesirable behaviour. Showing nothing when there are additional details missing doesn't make much sense to me. We have a useful summary present so it should be used.
